### PR TITLE
Update sample readme slug

### LINKED ISSUE #17994: [identity] update sample slug in sample README
The [sample slug](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/identity/azure-identity/samples) (the metadata under `products`) for azure-identity does not match any of the valid options found here:
https://review.docs.microsoft.com/en-us/help/contribute/metadata-taxonomies?branch=master#product

This needs to be updated so that [samples](https://docs.microsoft.com/en-us/samples/browse/?languages=python) publish correctly.

### PR DESCRIPTION
**Pull request by chlowell**
Closes #17994

### CODE REVIEW

File: sdk/identity/azure-identity/samples/README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ languages:
   - python
 products:
   - azure
-  - azure-identity
+  - azure-sdks
```
Comment (kristapratico) [inline sdk/identity/azure-identity/samples/README.md:7]: JS uses `azure-active-directory` here. Would it make sense to use the same?

Comment (chlowell) [reply sdk/identity/azure-identity/samples/README.md:7]: Would that cause these samples to show up in Azure AD documentation?

Comment (kristapratico) [reply sdk/identity/azure-identity/samples/README.md:7]: @witemple-msft do you know?

File: sdk/identity/azure-identity/samples/README.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ languages:
   - python
 products:
   - azure
-  - azure-identity
```
Comment (mccoyp) [inline sdk/identity/azure-identity/samples/README.md:7]: Is an `identity` tag necessary for these to show up with Identity samples?

File: sdk/identity/azure-identity/samples/README.md (hunk 1 above)
Comment (chlowell) [reply sdk/identity/azure-identity/samples/README.md:7]: Answered my own question: it does, in the [samples browser](https://docs.microsoft.com/en-us/samples/browse/?expanded=azure&products=azure-active-directory&languages=javascript) at least. At first glance I'm thinking our samples shouldn't be grouped with Azure Active Directory samples. This library uses Azure AD but is narrowly focused on authenticating Azure SDK clients. Our samples are poor demonstrations of Azure AD features and in general just noise for people not using the SDK.

File: sdk/identity/azure-identity/samples/README.md (hunk 2 above)
Comment (chlowell) [reply sdk/identity/azure-identity/samples/README.md:7]: I believe so but question whether we want these samples to show up there. Today there's no "there" there; the [sample browser](https://docs.microsoft.com/samples/browse/) has nothing for "identity". The sample browser covers products outside Azure, so not knowing the intended scope of "identity", I'm reluctant to make our samples the only ones in that category.

File: sdk/identity/azure-identity/samples/README.md (hunk 1 above)
Comment (kristapratico) [reply sdk/identity/azure-identity/samples/README.md:7]: Where would someone expect to find the identity SDK samples in the microsoft docs?

Comment (kristapratico) [reply sdk/identity/azure-identity/samples/README.md:7]: For example, in form recognizer, our MS docs menu has a drop down for "Samples" where our samples readme gets published: https://docs.microsoft.com/en-us/azure/cognitive-services/form-recognizer/overview?tabs=v2-1

Comment (chlowell) [reply sdk/identity/azure-identity/samples/README.md:7]: Authentication is covered in the [SDK conceptual docs](https://docs.microsoft.com/azure/developer/python/azure-sdk-authenticate). As for the sample browser, the taxonomy is ordered by product, so I think azure-identity samples belong under `azure` (though I think that's too broad a category to be really useful) and `azure-sdks`.

**Review by kristapratico — APPROVED** [review]

**Pull request merged**